### DISCUSSION TITLE
Can I install OS on initially unrecognized SSD with this script ?

### DISCUSSION THREAD

**Discussion by brupec** [007revad/Synology_HDD_db#342] [Q&A] [answered]
Hi all! 

have a DS923+ with 4 HDD which contains os.
I recently added a SSD which was not recognized.
I successfully used this script to get it recognized and get all my docker data on it, what a change (thanks a lot 007revad!)

But now I want to push it further, i'd like to take opportunity that I'm changing all the HDD to reinstall the OS on the SSD instead of the HDD raid (for faster response and less hdd wakes)

So I should start up with the ssd only in my NAS and install OS on it. Is there a way I will be able to run the script ? I imagine I won't be allow to install OS on SSD if there is only the unrecognized SSD in the NAS ? I imagine script won't be effective when I'll remove all HDD ?

Hope my question is clear ;-)

Cheers

**Comment by 007revad** [comment, marked as answer]
DSM doesn't install on an NVMe drive.

You'd need to start again with 2.5 inch SATA SSDs in the DS923+, then shut it down, install the NVMe drive, boot, run the script and do an online assemble of the storage pool on the NVMe drive.

I have one 2.5 inch SATA SSD and multiple HDDs. I ran the syno_hdd_db script with the -S option to set write_mostly for the HDDs so DSM will try to read from the SSD instead of the HDDs. Packages like Plex, docker and VMs are all on my NVMe drives.

**Reply by brupec** [reply]
Thanks a lot.

Quick question, what do you mean by "do an online assemble of the storage pool on the NVME drive" ?
Not sure of this step, and a bit scared to mess things up
I have to do it with a sata SSD necessarily, cannot use a HDD ?
After I do that, can I remove the sata SSD and keep only the nvme with OS/docker/etc on it and my HDD with my data ?

Can you also explain further this step of write_mostly for the hdd ? what does it change if the system is already only on SSD ? It should only read from there right ? Read the HDD just when accessing the files on it ?

**Reply by 007revad** [reply]
In DSM 7 "online assemble" appears in storage manager when DSM detects a drive or drives with a Synology storage pool and volume that aren't currently part of the current DSM setup.
https://github.com/007revad/Synology_M2_volume/blob/main/images/create_m2_volume_online_assemble.png
https://github.com/007revad/Synology_M2_volume/blob/main/images/create_m2_volume_online_assemble2.png

You can't install DSM on an NVMe drive. If you want DSM on an SSD drive you need a 2.5 inch SATA SSD. I would not bother because DSM only runs a tiny bit faster on an SSD.

write_mostly is only relevant when you have both HDDs and 2.5 inch SATA SSDs in the NAS. It makes DSM run from the 2.5 inch SATA SSDs instead of the HDDs.

Forget what you are trying to do and instead just: 
1. Install or move packages like Plex, docker and virtual machines to the NVMe drive.
2. Move the docker, PlexMediaServer and VMs shared folders to the NVMe drive.
3. Leave the 4 HDDs for data and DSM.

To move shared folders see https://github.com/007revad/Synology_app_mover/blob/main/images/move_shared_folder.png

To move packages see https://github.com/007revad/Synology_app_mover

**Reply by brupec** [reply]
Thanks a lot Dave for taking time to answer me.

Now all is more clear or what I can / can't do. Actually what you recommend is exactly what I did (except I didn't have your script, which would definately would have made my life much easier and save me much time...)

Also, I'm swapping the 4 hdd to bigger/more recent ones.

I'm thinking of putting the 4 new hdd directly and reinstall dsm, and restore backups (have to check how to do them, i have lot of dockers packages and would like to redo as less configuration as possible, but as most is on the SSD now if i'm not mistaken I shouldn't have too much work)
I have an old NAS where I'll put all old HDD and then transfer from old to new.

any recommendation on how to proceed smoothly ?

should I use synology control panel backup / hyperbackup / something else ?

**Reply by 007revad** [reply]
Backing up to a USB drive with hyper backup is fine.

If you are using RAID 5 or SHR and you want to keep the same RAID type I would just replace 1 drive then let storage manager repair the storage pool. Then when that's finished do the same for the 2nd drive. And repeat for the 3rd drive and then the 4th drive.

**Reply by brupec** [reply]
did backup in the interface, it didn't restore much but I didn't loose docker packages on the ssd, that's the most important. A bit of configuration and it's back. Now to wait the 8tb copy.
interesting fact is that even if I didn't yet run the script, my ssd volume was recognized as soon as I reinstalled dsm. Seems that when there's already synology stuff on the ssd, it reads it even if not in compatibility list.

anyway, I run the script, and also added support for my new hdds ;-)

thanks for all sir, take care
